### ISSUE TITLE
REAR Suse 11 SP4 running on IBM Power8 ISO image not bootable

### ISSUE THREAD

**Issue by suseuser168** [rear/rear#1375]
#### Relax-and-Recover (ReaR) Issue Template

Fill in the following items before submitting a new issue
(quick response is not guaranteed with free support):

* rear version (/usr/sbin/rear -V):Relax-and-Recover 1.17.1 / Git
* OS version (cat /etc/rear/os.conf or lsb_release -a): 
OS_VENDOR=SUSE_LINUX
OS_VERSION=11
* rear configuration files (cat /etc/rear/site.conf or cat /etc/rear/local.conf):
OUTPUT=ISO
BACKUP=NETFS
BACKUP_URL=file:///REAR_ISO/
ISO_MAX_SIZE=4500
ISO_DIR=/REAR_ISO
TMPDIR=/REAR_TEMP

* Are you using legacy BIOS or UEFI boot? BIOS boot
* Brief description of the issue: Running REAR on IBM Power8 hardware. Created a bootable backup ISO file using the command "rear -v mkbackup". Tried to boot up the ISO image using the IBM HMC in SMS mode and saw error with no bootable file found. How do I get this ISO image to be bootable?  

I tried to run the make rpm using the rear-master package to build the 2.00 version but I get errors saying :
== Building archive rear-2.00-git201706050000 ==
git checkout master
/bin/bash: git: command not found
make: *** [rear-2.00-git201706050000.tar.gz] Error 127

How do I get the latest already  built rpm for REAR which runs on IBM Power8? 
* Work-around, if any:

**Comment by jsmeix** [comment]
I am not a IBM Power user so that I can only provide
some general info, e.g. see
https://github.com/rear/rear/issues/1368#issuecomment-302330619
that reads (excerpts):
<pre>
Simply put:
ReaR 1.17.1 is too old to be used on IBM Power.

Recently there have been many adaptions and enhancements
from @schabrolles regarding ReaR on Power (ppc64/ppc64le)
</pre>

How to test the currently newest ReaR GitHub master code
(even independent of an already installed ReaR software):

Basically "git clone" it into a directory and then
configure and run it from within that directory like:
<pre>
# git clone https://github.com/rear/rear.git

# cd rear

# vi etc/rear/local.conf

# usr/sbin/rear -d -D mkbackup
</pre>
(note the relative paths "etc/rear/" and "usr/sbin/").

This means you must have 'git' installed.
On my SLES11 x86 system I have the "git-core" RPM package installed.
I don't know whether or not 'git' (e.g. the "git-core" RPM)
is provided via official SLES11 repositories.

If you cannot get 'git' installed on your SLES11 IBM Power system
it may also work when you go to
https://github.com/rear/rear
there click on the "Clone or download" button
and then use "Download ZIP"
which gets you a "rear-master.zip"
that contains the currentl ReaR GitHub master code.

When you 'unzip rear-master.zip' into a separated directory
and therein do "cd rear-master" it should work to configure
and run it from within that directory in the same way
as described above for a "git clone".
At least for me (not on IBM Power) it worked this way.

**Comment by suseuser168** [comment]
@jsmeix,
Thank you for your response and suggestions!

SUSE SLES 11 SP4 on Power doesn't include git-core. 

With the firewalls blocking outside access, I cannot run the git clone https://github.com/rear/rear.git.

I used your workaround and unzipped the rear-master.zip file into a directory ( /rear-master) and run REAR from that directory.

I imported the ISO file into the VIO server media repository and was able to boot up the ISO file using the IBM HMC in SMS mode but it doesn't have any data.

I am an AIX system admin trying to learn and use SUSE SLES 11 and 12 and ReaR so any help would be greatly appreciated.

Here are the files generated when I run "rear -d -D mkbackup"

total 1294128
drwxr-x--- 2 root root       4096 Jun  6 15:32 .
drwxr-xr-x 3 root root       4096 Jun  2 13:45 ..
-rw------- 1 root root        202 Jun  6 15:28 README
-rw------- 1 root root        270 Jun  6 15:28 VERSION
-rw-r--r-- 1 root root   12543360 Jun  6 15:32 backup.log
-rw------- 1 root root 1245151060 Jun  6 15:32 backup.tar.gz.00
-rw------- 1 root root   62615552 Jun  6 15:28 rear-suse.iso
-rw------- 1 root root    3413822 Jun  6 15:28 rear-suse.log
-rw------- 1 root root     130042 Jun  6 07:49 rear.log

I see separate backup.tar.gz.00 and ISO files.

Doesn't REAR combine the bootable image + data into one ISO file?

How do get it combined into 1 bootable ISO image with boot data and data?

I couldn't find any latest built rpm for SLES11SP4 and SLES12SP4 running on Power (ppc or ppcle).

Do you know where I can get the latest built rpm? 

I tried to run "make rpm" to build the rpm but got errors.

== Building archive rear-2.00-git201706060000 ==
git checkout master
/bin/bash: git: command not found
make: *** [rear-2.00-git201706060000.tar.gz] Error 127

**Comment by jsmeix** [comment]
ReaR can combine the bootable image + data into one ISO file,
see the example configs at usr/share/rear/conf/examples
in particular rescue-and-backup-on-same-ISO-image-example.conf

Right now I used in my local.conf (excerpt):
<pre>
OUTPUT=ISO
BACKUP=NETFS
BACKUP_OPTIONS="nfsvers=3,nolock"
BACKUP_URL=iso://backup
OUTPUT_URL=nfs://10.160.4.244/nfs
</pre>
and I got the backup in the ISO image:
<pre>
# usr/sbin/rear -d -D mkbackup
Relax-and-Recover 2.00 / Git
...
Creating recovery/rescue system initramfs/initrd initrd.cgz with gzip default compression
...
Creating tar archive '/tmp/rear.3nGa5SVAfztFvbF/tmp/isofs/backup/backup.tar.gz'
...
Archived 1416 MiB in 143 seconds [avg 10140 KiB/sec]
Making ISO image
Wrote ISO image: /root/rear-master.zip/rear-master/var/lib/rear/output/rear-e205.iso (1.5G)
Copying resulting files to nfs location
</pre>

With a huge backup this could cause disk-space issues
because on the system where "rear mkbackup" is run
the ISO image needs to be created and stored
<pre>
# ls -lh var/lib/rear/output/
total 1.5G
-rw------- 1 root root 1.5G Jun  7 11:38 rear-e205.iso
</pre>
and one needs all that space in ReaR's (temporary)
working area $TMPDIR/rear.XXXXXXXXXXXXXXX
where all the files exists wherefrom the ISO image
gets created.

This is what I get as final result on my NFS server:
<pre>
# ls -lh e205/
total 1.6G
-rw------- 1 nobody nogroup  202 Jun  7 11:39 README
-rw------- 1 nobody nogroup 1.5G Jun  7 11:39 rear-e205.iso
-rw------- 1 nobody nogroup 7.1M Jun  7 11:39 rear-e205.log
-rw------- 1 nobody nogroup  262 Jun  7 11:39 VERSION
</pre>

Booting the rear-e205.iso on another (replacement) machine
with same "hardware" ( I use QEMU/KVM virtual machines,
cf. "First steps with Relax-and-Recover" at
https://en.opensuse.org/SDB:Disaster_Recovery )
boots the ReaR recovery system and therein "rear recover"
recreates the system with the backup "inside" the ISO
(at least for me this "just works").

Regarding a latest built rpm for current ReaR master code:
We are currently in the process to release ReaR version 2.1
and when ReaR version 2.1 is released you can get RPMs
from the "usual places" as described
in "Version upgrades with Relax-and-Recover"
and "First steps with Relax-and-Recover" in
https://en.opensuse.org/SDB:Disaster_Recovery
BUT:
I think only RPMs for x86 architecture are available for SLES11.
I think for IBM Power no RPMs are available for SLES11.
In the end the architecture does not really matter for ReaR
because ReaR is only bash scripts.
In general why we provide RPMs only for certain architectures
is described in the RPM spec file, see
https://raw.githubusercontent.com/rear/rear/master/packaging/rpm/rear.spec

In general regarding IBM Power architecture see also
https://raw.githubusercontent.com/rear/rear/master/doc/rear-release-notes.txt
that reads (excerpt)
<pre>
Supported Architectures

Rear-2.1 is supported on:

  o Intel x86 type of processors
  o AMD x86 type of processors

Rear-2.1 may or may not fully work on:

  o Intel Itanium processors
  o PPC processors
  o PPC64 processors
  o PPC64LE processors

Rear-2.1 does not support:

  o ARM type of processors
  o s390x type of processors
</pre>

**Comment by schabrolles** [comment]
@suseuser168, 

- `git-core` is provided by the SLES11SP4-SDK DVD 
- If you need it, I can provide you rpm builded for ppc64 sles11, just tell me
- Regarding ISO with boot+backup, this option is currently not available for POWER.
I usually prefer to have an ISO included boot+tool and backup stored in a NFS server on the network (NETFS)

I put some example of this kind of configuration with rear on POWER in this presentation :  https://www.slideshare.net/sebastienchabrolles/relax-and-recover-on-power-updated-052017

**Comment by jsmeix** [comment]
@schabrolles 
many thanks for your help with POWE specific issues!

I wonder why ISO with included backup is not possible for POWER.
I assume it is because of
https://github.com/rear/rear/issues/1142
Or is there another reason?

**Comment by schlomo** [comment]
@schabrolles many thanks also from me and please also extend that to the other IBM colleagues who help with ReaR. I took the liberty to add your presentation to http://relax-and-recover.org/documentation/#external-links

**Comment by schabrolles** [comment]
thanks @schlomo @jsmeix 
Sure, I'll transmit your message to my to the IBM guys (@jayfurmanek and @mmitsugi) who worked to make ReaR work on POWER.

@jsmeix regarding the ISO+backup issue, I think it is related to #697. I'll try to have a look on it on my free time to evaluate what have to be done to make it work on Power.

**Comment by jsmeix** [comment]
@schabrolles 
thanks now even I see it! ;-)
The '*_create_iso*' scripts are architecure dependant.

On x86 we have three scripts
<pre>
output/ISO/Linux-i386/300_create_isolinux.sh
output/ISO/Linux-i386/800_create_isofs.sh
output/ISO/Linux-i386/820_create_iso_image.sh
</pre>
while on other architectures we have only one for each architecture
<pre>
output/ISO/Linux-ia64/800_create_isofs.sh

output/ISO/Linux-ppc64/800_create_isofs.sh

output/ISO/Linux-ppc64le/800_create_isofs.sh
</pre>

It seems in output/ISO/Linux-i386/820_create_iso_image.sh
the whole $TMP_DIR/isofs content gets included in the ISO
(and via things like BACKUP_URL=iso://backup the backup
becomes stored in $TMP_DIR/isofs/backup)
while in contrast e.g.
in output/ISO/Linux-ppc64/800_create_isofs.sh
only the files in the ISO_FILES array get included in the ISO
which explains why in https://github.com/rear/rear/issues/697
it helped to add $TMP_DIR/isofs to the ISO_FILES array.

Therefore on my first glance it seems all what is to do
is to adapt the ISO creation on the other architectures
to match how it is done on x86.

@schabrolles 
it seems after you adapted and cleaned up the bootloader install
scripts (together with the whole 'finalize' stage) for POWER
it looks now as if also the ISO making scripts also should
be adapted and cleaned up in a similar way...

**Comment by jsmeix** [comment]
@suseuser168 
according to your above
https://github.com/rear/rear/issues/1375#issuecomment-306612244
<pre>
I used your workaround and unzipped the rear-master.zip file
into a directory ( /rear-master) and run REAR from that directory.

I imported the ISO file into the VIO server media repository
and was able to boot up the ISO file
using the IBM HMC in SMS mode ...
</pre>
this particular issue here is solved so that I close it.

Regardless that it is closed one can still add furter comments
but I would prefer if for each separated topic a separated
GitHub issue is used to avoid that separated things get
mixed up into a huge incomprehensible mess.
